test/integration/podgroup_controller_test.go

```diff
@@ -0,0 +1,360 @@
+			},
+			expectedPGState: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Failed", "", 0, 2, 0, 1),
+			},
+		},
+		{
+			name: "The status of no podGroup label pod changes from pending to running",
+			podGroups: []*v1alpha1.PodGroup{
+				util.MakePG("pg1-1", ns, 3, nil, nil),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("t1-p1-1").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-2").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-3").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Obj(),
+			},
+			intermediatePGState: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Pending", "", 0, 0, 0, 0),
+			},
+			incomingPods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("t1-p1-1").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Phase(v1.PodRunning).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-2").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Phase(v1.PodRunning).Obj(),
+				st.MakePod().Namespace(ns).Name("t1-p1-3").Req(map[v1.ResourceName]string{v1.ResourceMemory: "50"}).Priority(
+					midPriority).Node(nodeName).Phase(v1.PodRunning).Obj(),
+			},
+			expectedPGState: []*v1alpha1.PodGroup{
+				util.UpdatePGStatus(util.MakePG("pg1-1", ns, 3, nil, nil), "Pending", "", 0, 0, 0, 0),
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			defer cleanupPodGroups(testCtx.Ctx, extClient, tt.podGroups)
+			defer cleanupPods(t, testCtx, tt.existingPods)
+			defer cleanupPods(t, testCtx, tt.incomingPods)
+			// create pod group
+			if err := createPodGroups(testCtx.Ctx, extClient, tt.podGroups); err != nil {
+				t.Fatal(err)
+			}
+
+			// create Pods
+			for _, pod := range tt.existingPods {
+				klog.InfoS("Creating pod ", "podName", pod.Name)
+				_, err := cs.CoreV1().Pods(pod.Namespace).Create(testCtx.Ctx, pod, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Failed to create Pod %q: %v", pod.Name, err)
+				}
+				if pod.Status.Phase == v1.PodRunning {
+					_, err = cs.CoreV1().Pods(pod.Namespace).UpdateStatus(testCtx.Ctx, pod, metav1.UpdateOptions{})
+					if err != nil {
+						t.Fatalf("Failed to update Pod status %q: %v", pod.Name, err)
+					}
+				}
+			}
+			if err := wait.Poll(time.Millisecond*200, 10*time.Second, func() (bool, error) {
+				for _, pod := range tt.incomingPods {
+					if !podScheduled(cs, ns, pod.Name) {
+						return false, nil
+					}
+				}
+				return true, nil
+			}); err != nil {
+				t.Fatalf("%v Waiting existPods create error: %v", tt.name, err.Error())
```

Comment:
```suggestion
				t.Fatalf("Waiting existPods %v create error: %v", tt.name, err.Error())
```

Reply:
Thanks for your comment!

tt.name is not pod name, but testname, so I think current comment is fine.